langchain4j-ollama/src/test/java/dev/langchain4j/model/ollama/OllamaStructuredOutputIT.java

```diff
@@ -0,0 +1,182 @@
+                        .type(ResponseFormatType.JSON)
+                        .jsonSchema(JsonSchema.builder()
+                                .rootElement(schema)
+                                .build())
+                        .build())
+                .build());
+
+        final String response = chatResponse.aiMessage().text();
+
+        // then
+        CountryInfo countryInfo = OllamaJsonUtils.toObject(response, CountryInfo.class);
+
+        assertThat(countryInfo.name()).isEqualTo("Canada");
+        assertThat(countryInfo.capital()).isEqualTo("Ottawa");
+        assertThat(countryInfo.languages()).contains("English", "French");
+    }
+
+    @Test
+    void should_generate_structured_output_streaming_response_format_way() throws ExecutionException, InterruptedException, TimeoutException {
+
+        // given
+        StreamingChatLanguageModel streamingOllamaChatModelWithResponseFormat = OllamaStreamingChatModel.builder()
+                .baseUrl(ollamaBaseUrl())
+                .modelName(TOOL_MODEL)
+                .temperature(0.0)
+                .responseFormat(ResponseFormat.builder().type(ResponseFormatType.JSON)
+                        .jsonSchema(JsonSchema.builder().rootElement(schema).build())
+                        .build())
+                .logRequests(true)
+                .logResponses(true)
+                .build();
+
+        // when
+        CompletableFuture<Response<AiMessage>> secondFutureResponse = new CompletableFuture<>();
+        streamingOllamaChatModelWithResponseFormat.generate("Tell me about Canada.", new StreamingResponseHandler<>() {
+
+            @Override
+            public void onNext(String token) {
+            }
+
+            @Override
+            public void onComplete(Response<AiMessage> response) {
+                secondFutureResponse.complete(response);
+            }
+
+            @Override
+            public void onError(Throwable error) {
+                secondFutureResponse.completeExceptionally(error);
+            }
+        });
+
+
+        // then
+        Response<AiMessage> secondResponse = secondFutureResponse.get(30, SECONDS);
+        AiMessage secondAiMessage = secondResponse.content();
+        CountryInfo countryInfo = OllamaJsonUtils.toObject(secondAiMessage.text(), CountryInfo.class);
+
+        assertThat(countryInfo.name()).isEqualTo("Canada");
+        assertThat(countryInfo.capital()).isEqualTo("Ottawa");
+        assertThat(countryInfo.languages()).contains("English", "French");
+    }
+
+    @Test
+    void should_generate_structured_output_streaming_old_format_way() throws ExecutionException, InterruptedException, TimeoutException {
+        // given
+        StreamingChatLanguageModel streamingOllamaChatModelWithOldFormatWayJsonSchema = OllamaStreamingChatModel.builder()
+                .baseUrl(ollamaBaseUrl())
+                .modelName(TOOL_MODEL)
+                .temperature(0.0)
+                .format(OllamaJsonUtils.toJson(
```

Comment:
this should be .format("json") instead

Reply:
yes, correct. This is a deprecation, so we should support only what was supported so far. For the new feature they need ot use the new API

Reply:
Okay, so I’ll delete that test/example since it doesn’t make any sense.